aioauth/grant_type.py

```diff
@@ -166,6 +166,10 @@ class PasswordGrantType(GrantTypeBase[TRequest, TStorage]):
     """
 
     async def validate_request(self, request: TRequest) -> Client:
+        # Password grant requires a client_secret
+        if self.client_secret is None:
+            raise InvalidClientError(request, "")
```

Comment:
second argument (description) is not needed here, also `TRequest` type is required here:

```suggestion
            raise InvalidClientError[TRequest](request)
```